src/types.jl

```diff
@@ -211,11 +211,11 @@ mutable struct Geometry <: AbstractGeometry
     Geometry(ptr::GDALGeometry = C_NULL) = new(ptr)
 end
 
-mutable struct IGeometry <: AbstractGeometry
+mutable struct IGeometry{OGRwkbGeometryType} <: AbstractGeometry
```

Comment:
It's a mutable struct, are there instances where we change (from a NULL pointer) to a new geometry type?

Reply:
Not that I know off.

Reply:
Not that I know of either.

But I'm curious might the `OGRwkbGeometryType` in the type potentially drift from the underlying pointer? I.e. that `typeof(geom) != ArchGDAL.getgeomtype(geom)` per https://github.com/yeesian/ArchGDAL.jl/blob/f56f79c23b47e43d968605b053a531cf902991ae/src/ogr/geometry.jl#L301

It seems that might be possible with the `ArchGDAL.forceto` method: https://github.com/yeesian/ArchGDAL.jl/blob/f56f79c23b47e43d968605b053a531cf902991ae/src/ogr/geometry.jl#L138-L145

Reply:
Circling back on this: I'm guessing in those cases we could return a new geometry of the correspond type, so that isn't an issue.